tests/test_config/test_correct_year.py

```diff
@@ -1,29 +1,172 @@
+    if expect_date:
+        assert cfg.copyright == f'2006-{expect_date}'
+    else:
+        assert cfg.copyright == copyright_date
+
+
+def test_correct_year_single(expect_date):
+    # test that copyright is substituted
+    copyright_date = '2009, Alice'
+    cfg = Config(dict(copyright=copyright_date), {})
+    assert cfg.copyright == copyright_date
+    correct_copyright_year(None, cfg)
+    if expect_date:
+        assert cfg.copyright == f'{expect_date}, Alice'
+    else:
+        assert cfg.copyright == copyright_date
+
+
+def test_correct_year_single_space(expect_date):
+    # test that copyright is substituted
+    copyright_date = '2009 Alice'
+    cfg = Config(dict(copyright=copyright_date), {})
+    assert cfg.copyright == copyright_date
+    correct_copyright_year(None, cfg)
+    if expect_date:
+        assert cfg.copyright == f'{expect_date} Alice'
+    else:
+        assert cfg.copyright == copyright_date
+
+
+def test_correct_year_single_no_author(expect_date):
+    # test that copyright is substituted
+    copyright_date = '2009'
+    cfg = Config(dict(copyright=copyright_date), {})
+    assert cfg.copyright == copyright_date
+    correct_copyright_year(None, cfg)
+    if expect_date:
+        assert cfg.copyright == f'{expect_date}'
+    else:
+        assert cfg.copyright == copyright_date
+
+
+def test_correct_year_multi_line(expect_date):
+    # test that copyright is substituted
+    copyright_dates = (
+        '2006',
+        '2006-2009, Alice',
+        '2010-2013, Bob',
+        '2014-2017, Charlie',
+        '2018-2021, David',
+        '2022-2025, Eve',
+    )
+    cfg = Config(dict(copyright=copyright_dates), {})
+    assert cfg.copyright == copyright_dates
+    correct_copyright_year(None, cfg)
+    if expect_date:
+        assert cfg.copyright == (
+            f'{expect_date}',
+            f'2006-{expect_date}, Alice',
+            f'2010-{expect_date}, Bob',
+            f'2014-{expect_date}, Charlie',
+            f'2018-{expect_date}, David',
+            f'2022-{expect_date}, Eve',
+        )
+    else:
+        assert cfg.copyright == copyright_dates
+
+
+def test_correct_year_multi_line_all_formats(expect_date):
+    # test that copyright is substituted
+    copyright_dates = (
+        '2009',
+        '2009 Alice',
+        '2009, Bob',
+        '2006-2009',
+        '2006-2009 Charlie',
+        '2006-2009, David',
+    )
+    cfg = Config(dict(copyright=copyright_dates), {})
+    assert cfg.copyright == copyright_dates
+    correct_copyright_year(None, cfg)
+    if expect_date:
+        assert cfg.copyright == (
+            f'{expect_date}',
+            f'{expect_date} Alice',
+            f'{expect_date}, Bob',
+            f'2006-{expect_date}',
+            f'2006-{expect_date} Charlie',
+            f'2006-{expect_date}, David',
+        )
+    else:
+        assert cfg.copyright == copyright_dates
+
+
+def test_correct_year_app(expect_date, tmp_path, make_app):
+    # integration test
```

Comment:
Of interest, pycharm reports that this single test takes ~471ms. All 45 tests in the file take 473ms -- goes to show that we should prefer unit tests when possible!